libsvmdata/tests/test_datasets.py

```diff
@@ -1,45 +1,65 @@
+import pytest
 import numpy as np
 
 from libsvmdata import fetch_libsvm
 
+TEST_DATASETS = {
+    "regression": [
+        ("abalone", 4_177, 8),
+        ("bodyfat", 252, 14),
+    ],
+    "binary": [
+        ("a1a", 1_605, 123),
+        ("breast-cancer", 683, 10),
+    ],
+    "multiclass": [
+        ("dna", 2_000, 180),
+        ("iris", 150, 4),
+    ],
+    "multilabel": [
+        ("bibtex", 7_395, 1_836),
+        ("scene-classification", 1_211, 294),
+    ],
+}
 
-# TODO: test all datasets without timeout
-# @pytest.mark.parametrize("name", dataset.NAMES.keys())
-# def test_datasets(name):
-#     if "sector" in name:
-#         pytest.xfail(name)
+@pytest.mark.filterwarnings("ignore:FutureWarning")
+@pytest.mark.parametrize("dataset_name,n,p", TEST_DATASETS["regression"])
+def test_regression(dataset_name, n, p):
+    X, y = fetch_libsvm(dataset_name)
+    assert X.shape[0] == n
```

Comment:
to remain consistent with ``test_core``, use testing utils of ``numpy``

Reply:
In `test_core`, I use `numpy` utilities to compare the equality between two `np.array`. Here, I only assert the equality between integers so I'm not sure this is a good option to use `numpy` testing utilities. However, if it really matters, I can adapt my code without any problem !

Reply:
As a rule of thumb I prefer numpy because it raises informative error messages (here assert_equal would print the actual and desired values in case of mismatch). IMO it's worth the slightly heavier syntax. 
Also, I'm fine with import some `assert_` functions directly instead of using `np.testing.assert_`

Reply:
Ok, I've switched to `numpy`